File: flutter_app/lib/models/gym.dart
import 'package:json_annotation/json_annotation.dart';

part 'gym.g.dart';

@JsonSerializable()
class Gym {
  final String id;
  @JsonKey(name: 'name_en')
  final String nameEn;
  @JsonKey(name: 'name_ar')
  final String? nameAr;
  @JsonKey(name: 'description_en')
  final String? descriptionEn;
  @JsonKey(name: 'description_ar')
  final String? descriptionAr;
  final String tier;
  final String address;
  final double lat;
  final double lng;
  final String? phone;
  @JsonKey(name: 'logo_url')
  final String? logoUrl;
  @JsonKey(name: 'cover_url')
  final String? coverUrl;
  final List<dynamic>? photos;
  @JsonKey(name: 'opening_hours')
  final Map<String, dynamic>? openingHours;
  final List<String>? amenities;
  final List<String>? categories;
  @JsonKey(name: 'is_active')
  final bool isActive;
  @JsonKey(name: 'is_featured')
  final bool isFeatured;
  final double rating;
  @JsonKey(name: 'total_reviews')
  final int totalReviews;
  @JsonKey(name: 'country_id')
  final int countryId;

  const Gym({
    required this.id,
    required this.nameEn,
    this.nameAr,
    this.descriptionEn,
    this.descriptionAr,
    required this.tier,
    required this.address,
    required this.lat,
    required this.lng,
    this.phone,
    this.logoUrl,
    this.coverUrl,
    this.photos,
    this.openingHours,
    this.amenities,
    this.categories,
    this.isActive = true,
    this.isFeatured = false,
    this.rating = 0.0,
    this.totalReviews = 0,
    required this.countryId,
  });

  String name(String lang) => lang == 'ar' ? (nameAr ?? nameEn) : nameEn;

  factory Gym.fromJson(Map<String, dynamic> json) => _$GymFromJson(json);
  Map<String, dynamic> toJson() => _$GymToJson(this);
}

File: backend/app/models/gym.py
import uuid
from sqlalchemy import Column, String, Integer, Boolean, DECIMAL, DateTime, Text, func
from sqlalchemy.dialects.postgresql import UUID, JSONB
from app.database import Base


class Gym(Base):
    __tablename__ = "gyms"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    partner_id = Column(UUID(as_uuid=True), nullable=True)
    country_id = Column(Integer, nullable=False)
    name_en = Column(String(255), nullable=False)
    name_ar = Column(String(255), nullable=True)
    description_en = Column(Text, nullable=True)
    description_ar = Column(Text, nullable=True)
    tier = Column(String(20), nullable=False, default="standard")
    address = Column(String(500), nullable=False)
    city_id = Column(Integer, nullable=True)
    lat = Column(DECIMAL(10, 7), nullable=False)
    lng = Column(DECIMAL(10, 7), nullable=False)
    phone = Column(String(20), nullable=True)
    logo_url = Column(Text, nullable=True)
    cover_url = Column(Text, nullable=True)
    photos = Column(JSONB, default=[])
    opening_hours = Column(JSONB, nullable=False)
    amenities = Column(JSONB, default=[])
    categories = Column(JSONB, default=[])
    is_active = Column(Boolean, nullable=False, default=False)
    is_featured = Column(Boolean, nullable=False, default=False)
    rating = Column(DECIMAL(2, 1), default=0.0)
    total_reviews = Column(Integer, default=0)
    max_daily_visits = Column(Integer, default=1)
    deleted_at = Column(DateTime(timezone=True), nullable=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), nullable=False)
